hudi-common/src/main/java/org/apache/hudi/common/table/read/KeyBasedFileGroupRecordBuffer.java

```diff
@@ -77,6 +79,11 @@ public void processDataBlock(HoodieDataBlock dataBlock, Option<KeySpec> keySpecO
     try (ClosableIterator<T> recordIterator = recordsIteratorSchemaPair.getLeft()) {
       while (recordIterator.hasNext()) {
         T nextRecord = recordIterator.next();
+        String recordKey = readerContext.getRecordKey(nextRecord, schema);
+        // The filter is given but record key does not pass the filter.
+        if (useFilter && !evalFilter.get().apply(recordKey)) {
```

Comment:
instead of requiring a `useFilter` arg, can we just have a filter that always returns true?

Reply:
There is unnecessary overhead `readerContext.getRecordKey(nextRecord, schema)` if the `userFilter` is false.